### PR TITLE
[MRESOLVER-326] Introduce retries on HTTP connection errors

### PR DESCRIPTION
**Pull request by jerrinot**
My attempt to fix https://issues.apache.org/jira/browse/MRESOLVER-326

Feedback appreciated!

### CODE REVIEW

**Comment by cstamas** [comment]
Formatting issues: `mvn spotless:apply`, commit and push pls

File: maven-resolver-transport-http/src/test/java/org/eclipse/aether/transport/http/HttpServer.java [hunk 1]
```diff
@@ -109,6 +110,8 @@ public enum ChecksumHeader {
 
     private String proxyPassword;
 
+    private int connectionsToClose = 0;
```
[the remark this answers is not in the document]
Comment (jerrinot) [reply maven-resolver-transport-http/src/test/java/org/eclipse/aether/transport/http/HttpServer.java:113]: good one, thanks!

File: maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java [hunk 2]
```diff
@@ -129,6 +129,18 @@ public final class ConfigurationProperties {
      */
     public static final String DEFAULT_HTTP_CREDENTIAL_ENCODING = "ISO-8859-1";
 
+    /**
+     * The maximum number of times a request to a remote server should be retried in case of an error.
+     *
+     * @see #DEFAULT_HTTP_RETRY_HANDLER_COUNT
+     */
+    public static final String HTTP_RETRY_HANDLER_COUNT = PREFIX_CONNECTOR + "http.retryHandler.count";
```
Comment (slawekjaranowski) [inline maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java:137]: New property should also be added to `src/site/markdown/configuration.md`

Comment (cstamas) [reply maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java:137]: that renders here (just FTR) https://maven.apache.org/resolver/configuration.html

Comment (jerrinot) [reply maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java:137]: thanks to both of you!

I added a new row & copied a property description from the code. but I am not sure if the description fits the rest of the properties. it feels a bit off in style.

**Review by cstamas — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jerrinot** [comment]
Thanks to both @cstamas and @slawekjaranowski for your super quick help! I wish all my PRs were merged as quickly as this one! :)

**Comment by jira-importer** [comment]
Resolve #1000

**Pull request merged**